commit 62c4cf83231254d63f7bb205fc8743591072c9ce
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Thu May 18 15:36:46 2017 +0100

    staging: bcm2835-audio: Fix memory corruption
    
    I'm all for fixing memory leaks, but freeing a block while it is still
    being used is a recipe for hard-to-debug kernel exeptions.
    
    1) There is already a vchi method for freeing the instance, so use it.
    2) Only call it on error, and then only before initted is false.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>
    Fixes: 0adbfd4694c2 ("staging: bcm2835-audio: fix memory leak in bcm2835_audio_open_connection()")

diff --git a/drivers/staging/vc04_services/bcm2835-audio/bcm2835-vchiq.c b/drivers/staging/vc04_services/bcm2835-audio/bcm2835-vchiq.c
index 5f3d8f2339e3..89f96f3c0280 100644
--- a/drivers/staging/vc04_services/bcm2835-audio/bcm2835-vchiq.c
+++ b/drivers/staging/vc04_services/bcm2835-audio/bcm2835-vchiq.c
@@ -409,6 +409,7 @@ static int bcm2835_audio_open_connection(struct bcm2835_alsa_stream *alsa_stream
 			LOG_ERR("%s: failed to connect VCHI instance (ret=%d)\n",
 				__func__, ret);
 
+			vchi_disconnect(vchi_instance);
 			ret = -EIO;
 			goto err_free_mem;
 		}
@@ -431,7 +432,6 @@ static int bcm2835_audio_open_connection(struct bcm2835_alsa_stream *alsa_stream
 	LOG_DBG(" success !\n");
 	ret = 0;
 err_free_mem:
-	kfree(vchi_instance);
 
 	return ret;
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1037_62c4cf83231254d63f7bb205fc8743591072c9ce.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1037_62c4cf83231254d63f7bb205fc8743591072c9ce.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
